types/account.go

```diff
@@ -25,144 +21,228 @@ const (
 	Bech32PrefixValPub  = "cosmosvalpub"
 )
 
-// Bech32ifyAcc takes Address and returns the bech32 encoded string
-func Bech32ifyAcc(addr Address) (string, error) {
-	return bech32.ConvertAndEncode(Bech32PrefixAccAddr, addr.Bytes())
+//__________________________________________________________
+
+// AccAddress is a go crypto-style Address
```

Comment:
Mhmmm, seems like we could improve the explanation here? Thoughts? What exactly is an "Account Address" and what is it used for?